addons/sale_loyalty/models/sale_order.py

```diff
@@ -798,16 +798,19 @@ def _update_programs_and_rewards(self):
         if point_entries_to_unlink:
             point_entries_to_unlink.sudo().unlink()
 
+    def _get_not_rewarded_order_lines(self):
+        return self.order_line.filtered(lambda line: line.product_id and not line.reward_id)
+
     def _program_check_compute_points(self, programs):
         """
         Checks the program validity from the order lines aswell as computing the number of points to add.
 
         Returns a dict containing the error message or the points that will be given with the keys 'points'.
         """
         self.ensure_one()
-
         # Prepare quantities
-        order_lines = self.order_line.filtered(lambda line: line.product_id and not line.reward_id)
+        order_lines = self._get_not_rewarded_order_lines()
+
```

Comment:
The comment above is also for the next lines, so it's better to not separate these =]